### ISSUE TITLE
clean task deletes all sources...

### ISSUE THREAD

**Issue by amirabiri-odoro** [etiennestuder/gradle-jooq-plugin#60]
I place my generated files inside src and commit them to version control. So I have a config like this:

    jooq {
        main {
            ...
            generator {
                ...
                target {
                    packageName = 'com.company.service.db'
                    directory   = 'src/main/java'
                }
            }
        }
    }

This way my files are generated in `src/main/java/com/company/service/db`

However when I run the clean task, which triggers this plugin's `cleanGenerateMainJooqSchemaSource` task, it deletes `src/main/java`, which isn't desireable..

**Comment by etiennestuder** [comment]
See the full thread on this issue: https://github.com/etiennestuder/gradle-jooq-plugin/issues/59

**Comment by amirabiri-odoro** [comment]
This is incorrect. The clean task shouldn't delete `{directory}`. It should delete `{directory}/{package}`.

**Comment by etiennestuder** [comment]
The other issues explains the reasoning in detail.

**Comment by amirabiri-odoro** [comment]
OK I see your point.

To be honest, I removed the connection between build and jooq generation. Our workflow is that the developer runs the jooq generation task manually when they add migrations. The new deletion task surprised me because I upgraded the plugin from `1.0.6` to `2.0.9` and the clean task appeared suddenly and deleted everything...

With this workflow the rules RE Gradle source sets arguably don't apply. Furthermore We can't use the automatic generate workflow because we run builds on a build server that doesn't have a DB connected.

With that in mind I would say two things:
1. I think the `remove_task_dependency` should be updated to also demonstrate removal of the dependency between the clean tasks.
2. Should this workflow be officially supported? i.e perhaps the plugin's DSL can have a switch for running or not running the jooq generation/clean tasks automatically?
